### PR TITLE
treesitter: Update to 0.19.3

### PR DESCRIPTION
**Pull request by mfussenegger**
Based on https://github.com/neovim/neovim/pull/14045
Also updates the c parser and the highlights to get the tests passing.

There were some comments about removing the highlights in the other PR, but I
feel like that is an independent action unrelated to updating the TS version.

### CODE REVIEW

**Review by teto — APPROVED** [review]

**Comment by vigoux** [comment]
While I agree we should update, we should not consider tree-sitter 0.19 as a regular update.
It introduces breaking changes in the language ABI, and thus all parsers that we currently more or less "support" (through nvim-treesitter mainly), will need to be updated at once for all nightly users.

We are currently planning this move in nvim-treesitter, we just need a bit more time :wink:

File: runtime/queries/c/highlights.scm
Comment (vigoux) [inline runtime/queries/c/highlights.scm:130]: I think that the `has-ancestor`predicate is only implemented in nvim-treesitter and not in here, so that needs to be removed.

**Review by vigoux — CHANGES_REQUESTED** [review]

**Comment by teto** [comment]
> We are currently planning this move in nvim-treesitter, we just need a bit more time wink

link to the discussion ? I am curious as how it's going to be handled. would older neovim get served older grammar for instance ?

**Comment by vigoux** [comment]
The thing is that the runtime version and the parsers verion need to match, so if you use an "older" version of neovim you'll _have_ to use an "older" version of the parsers.

File: runtime/queries/c/highlights.scm
Comment (mfussenegger) [reply runtime/queries/c/highlights.scm]: Done :+1:

**Comment by maxbrunsfeld** [comment]
It might be good to include https://github.com/tree-sitter/tree-sitter/pull/971 in this PR, because it fixes several bugs that were reported in the context of neovim. I plan to merge it within the hour.

**Comment by mfussenegger** [comment]
Updated the PR to include 0.19.3

**Comment by tjdevries** [comment]
(closed my PR, we can just do this one whenever @vigoux is ready)

**Comment by tjdevries** [comment]
( We're just waiting on vigoux sign off ? )

**Comment by stsewd** [comment]
https://github.com/neovim/neovim/pull/14039 should be merged before this one, as some parsers will fail after upgrading to >=0.18.x

**Comment by theHamsta** [comment]
@stsewd that's not true. #14039 is only for the internals of our query linter. It does not affect users. I'd say this is good to go when @vigoux agrees.

**Pull request merged**
